File: social-media-react-main/src/models/Post.ts
import Comments from "./Comments";

export default class Post {
    id: number;
    text: string;
    imageUrl: string;
    comments: Comments[];
    author: any;
    likes: number[];

    constructor (id: number, text:string, imageUrl:string, comments: Comments[], author:any, likes:number[]) {
        this.id = id;
        this.text = text;
        this.imageUrl = imageUrl;
        this.comments = comments;
        this.author = author;
        this.likes = likes;
    }
}

File: social-media-spring-main/src/main/java/com/revature/models/Post.java
package com.revature.models;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;
import org.springframework.data.annotation.CreatedDate;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "posts")
public class Post {

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
	private String text;
	private String imageUrl;
	@CreatedDate
	private Date createdAt;

	@ElementCollection(fetch = FetchType.EAGER)
	@CollectionTable(name = "likes", joinColumns = @JoinColumn(name = "post_id"))
	private List<Integer> likes = new ArrayList<>();

    @JsonIgnore
	@OneToMany(cascade = CascadeType.ALL, mappedBy = "post")
	private List<Comments> comments;

	@ManyToOne
	private User author;



}
